src/items/extern-crates.md

```diff
@@ -38,6 +38,50 @@ Here is an example:
 extern crate hello_world; // hyphen replaced with an underscore
 ```
 
-[RFC 940]: https://github.com/rust-lang/rfcs/blob/master/text/0940-hyphens-considered-harmful.md
+## Extern Prelude
+
+All external crates are available in the "extern prelude" which exposes the
+crate names into lexical scope of every module without the need for specifying
+`extern crate`.
+
+> **Edition Differences**: In the 2015 edition, crates in the extern prelude
+> cannot be referenced via [use declarations], so it is generally standard
+> practice to include `extern crate` declarations to bring them into scope.
+>
+> Beginning in the 2018 edition, [use declarations] can reference crates in
+> the extern prelude, so it is considered unidiomatic to use `extern crate`.
+
+> **Note**: Additional crates that ship with `rustc`, such as [`proc_macro`],
+> [`alloc`], and [`test`], currently aren't available in the extern prelude
+> and must be brought into scope with an `extern crate` declaration, even in
+> the 2018 edition. `use` paths must reference the `extern crate` item (such
+> as using [`crate::`] or [`self::`] path prefixes).
+>
+> ```rust
+> extern crate proc_macro;
+> // Cannot reference `proc_macro` directly because it is not in the extern prelude.
+> // use proc_macro::TokenStream;
+> // Instead, you must reference the item in scope from the `extern crate`
+> // declaration.
+> use self::proc_macro::TokenStream;
+> ```
+
+<!--
+Possible upcoming changes that will change this:
+
+`extern crate` items will automatically be added to the extern prelude.
+    https://github.com/rust-lang/rust/pull/54658
```

Comment:
This is merged now

Reply:
The changes are behind the `extern_crate_item_prelude` feature flag, so there's not much difference.  I don't know what the future holds, there are no tracking issues. I tweaked the wording slightly, but it still seems a little awkward to me.  Is the distinction between the `--extern` added to extern prelude and `extern crate` being placed as an item in the current scope clear enough?  Should it emphasize this distinction more?